controllers/hetznercluster_controller_test.go

```diff
@@ -840,3 +842,100 @@ var _ = Describe("reconcileRateLimit", func() {
+		got := SetControlPlaneEndpoint(hetznerCluster)
+		if (hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4 != "xyz"){
+			t.Fatalf("Wrong input for hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4. Got: %s, Want: 'xyz'", hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4)
+		}
+		if (hetznerCluster.Spec.ControlPlaneLoadBalancer.Port != 12){
+			t.Fatalf("Wrong input for hetznerCluster.Spec.ControlPlaneLoadBalancer.Port. Got: %d, Want: 12", hetznerCluster.Spec.ControlPlaneLoadBalancer.Port)
+		}
+		assert.Equal(t, true, got)
+	})
+	t.Run("function should return true if hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled is true, hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4 is not nil and hetznerCluster.Spec.ControlPlaneEndpoint is nil", func(t *testing.T) {
+		hetznerCluster := &infrav1.HetznerCluster{Spec: infrav1.HetznerClusterSpec{ControlPlaneLoadBalancer: infrav1.LoadBalancerSpec{Enabled: true, Port: 11}, ControlPlaneEndpoint: nil}, Status: infrav1.HetznerClusterStatus{ControlPlaneLoadBalancer: &infrav1.LoadBalancerStatus{IPv4: "xyz"}}}
+		got := SetControlPlaneEndpoint(hetznerCluster)
+		if (hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4 != "xyz"){
+			t.Fatalf("Wrong input for hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4. Got: %s, Want: 'xyz'", hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4)
+		}
+		// value for ControlPlaneEndpoint (Host and Port) will change after SetControlPlaneEndpoint() is run
+		if (hetznerCluster.Spec.ControlPlaneEndpoint == nil){
+			t.Fatal("Value of ControlPlaneEndpoint should have been changed. It should not remain nil")
+		}
+		// Values of hetznerCluster.Spec.ControlPlaneEndpoint.Host and hetznerCluster.Spec.ControlPlaneEndpoint.Port should change after execution of the function SetControlPlaneEndpoint()
+		// They should be the same as hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4 for Host (Spec.ControlPlaneEndpoint.Host) and hetznerCluster.Spec.ControlPlaneLoadBalancer.Port for Port (Spec.ControlPlaneEndpoint.Port)
+		if (hetznerCluster.Spec.ControlPlaneEndpoint.Host != hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4){
+			t.Fatalf("Wrong value for Host set. Got: %s, Want: %s", hetznerCluster.Spec.ControlPlaneEndpoint.Host, hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4)
+		}
+		if (hetznerCluster.Spec.ControlPlaneEndpoint.Port != int32(hetznerCluster.Spec.ControlPlaneLoadBalancer.Port)){
+			t.Fatalf("Wrong value for Port set. Got: %d, Want: %d", hetznerCluster.Spec.ControlPlaneEndpoint.Port, int32(hetznerCluster.Spec.ControlPlaneLoadBalancer.Port))
+		}
+		assert.Equal(t, true, got)
+	})
+	t.Run("function should return true if hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled is true, hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4 is not nil, hetznerCluster.Spec.ControlPlaneEndpoint.Host is an empty string and hetznerCluster.Spec.ControlPlaneEndpoint.Port is 0", func(t *testing.T) {
```

Comment:
same thing. Include the changes (or no changes) that you expect for ControlPlaneEndpoint